### PR TITLE
Fix java.lang.UnsupportedOperationException: JsonNull in percentile aggs

### PR DESCRIPTION
**Pull request by avifro-dev**
Since we added the percentile aggregation we keep getting UnsupportedOperationException in case of the field we try to aggregate on is missing. This fix should fix these errors.

### CODE REVIEW

File: jest-common/src/main/java/io/searchbox/core/search/aggregation/PercentilesAggregation.java [hunk 1]
```diff
@@ -23,7 +25,8 @@ public PercentilesAggregation(String name, JsonObject percentilesAggregation) {
 
     private void parseSource(JsonObject source) {
         for (Map.Entry<String, JsonElement> entry : source.entrySet()) {
-            if(!(Double.isNaN(entry.getValue().getAsDouble()))) {
+            // In case of no results, in Elasticsearch the behaviour was to get 'NaN' and in OpenSearch is 'null'
+            if(!entry.getValue().equals(jsonNull) && !(Double.isNaN(entry.getValue().getAsDouble()))) {
```
Comment (avifro-dev) [inline jest-common/src/main/java/io/searchbox/core/search/aggregation/PercentilesAggregation.java:29]: We could do just `instance of`, but I thought it'll be nicer.
That's the `equals()` method of `JsonNull`
image

Comment (w0jt) [reply jest-common/src/main/java/io/searchbox/core/search/aggregation/PercentilesAggregation.java:29]: i'd compare the `entry` directly to `JsonNull.INSTANCE` which makes it clear that it's a singleton (imo the `jsonNull` variable here is unnecessary and makes it less obvious)

**Review by w0jt — APPROVED** [review]
🚀

File: jest-common/src/main/java/io/searchbox/core/search/aggregation/PercentilesAggregation.java (hunk 1 above)
Comment (avifro-dev) [reply jest-common/src/main/java/io/searchbox/core/search/aggregation/PercentilesAggregation.java:29]: agree.. will change

**Comment by avifro-dev** [comment]
Can't push new commits for now .
on it
image

**Pull request closed without merging**